lgc/patch/PatchResourceCollect.cpp

```diff
@@ -1712,37 +1721,48 @@ void PatchResourceCollect::matchGenericInOut() {
     assert(inOutUsage.outputMapLocCount == 0);
     bool generatingColorExportShader = m_shaderStage == ShaderStageFragment;
     generatingColorExportShader &= m_pipelineState->isUnlinked() && !m_pipelineState->hasColorExportFormats();
-    for (auto locMapIt = outLocMap.begin(); locMapIt != outLocMap.end();) {
-      auto &locMap = *locMapIt;
+
+    for (auto locInfoMapIt = outLocInfoMap.begin(); locInfoMapIt != outLocInfoMap.end();) {
+      const unsigned origLocation = locInfoMapIt->first.getLocation();
+      auto &newLocationInfo = locInfoMapIt->second;
+
       if (m_shaderStage == ShaderStageFragment) {
-        unsigned location = locMap.first;
         if (!generatingColorExportShader &&
-            m_pipelineState->getColorExportFormat(location).dfmt == BufDataFormatInvalid) {
-          locMapIt = outLocMap.erase(locMapIt);
+            m_pipelineState->getColorExportFormat(origLocation).dfmt == BufDataFormatInvalid) {
+          locInfoMapIt = outLocInfoMap.erase(locInfoMapIt);
           continue;
         }
       }
 
-      if (m_shaderStage == ShaderStageGeometry) {
-        if (locMap.second == InvalidValue) {
-          unsigned outLocInfo = locMap.first;
-          mapGsGenericOutput(*(reinterpret_cast<InOutLocationInfo *>(&outLocInfo)));
-        }
+      if (m_shaderStage == ShaderStageGeometry && newLocationInfo.isInvalid()) {
```

Comment:
previously the check for invalid for geometry shaders was inside this if statement, so nothing would be done for non-invalid location infos. Moving the invalid check here means the 'else' below will be run for valid locations - is that intended? Seems like the only change is that inOutUsage.outputMapLocCount could be changed

Reply:
Right, the logic of the previous version is clearer. I have restored. Thanks.